src/js/api/config.js

```diff
@@ -61,9 +61,50 @@ function _adjustDefaultBwEstimate(estimate) {
     return Defaults.bandwidthEstimate;
 }
 
+function _copyToLocalization(allOptions) {
+    const { advertising, related, sharing } = allOptions;
+    const localization = Object.assign({}, Defaults.localization, allOptions.localization);
+
+    if (advertising) {
+        localization.advertising = Object.assign({}, localization.advertising, {
+            admessage: advertising.admessage,
+            cuetext: advertising.cuetext,
+            loadingAd: localization.loadingAd,
+            podmessage: advertising.podmessage,
+            skipmessage: advertising.skipmessage,
+            skiptext: advertising.skiptext
+        });
+    }
+
+    if (related) {
+        if (typeof localization.related === 'object') {
+            localization.related = Object.assign({}, localization.related, {
+                autoplaymessage: related.autoplaymessage,
+                heading: localization.nextUp || related.heading,
+            });
+        } else {
+            localization.related = Object.assign({}, Defaults.localization.related, {
+                heading: localization.related
+            });
+        }
+    }
+
+    if (sharing) {
+        localization.sharing = Object.assign({}, localization.sharing, {
```

Comment:
the problem here is that if the developer specified both in the localization block and in the sharing block, the localization block should take precedence;
instead i would write
```
localization.sharing.heading = localization.sharing.heading || sharing.heading;
localization.sharing. copied = localization.sharing. copied || localization. copied;
```